### PR TITLE
Load one file for the list of paths

### PR DESCRIPTION
**Pull request by fiskus**
API for the `fetchFile` is preserved:

```ts
fetchFile({ s3, bucket, path: 'path-to-the-file' })
```

with additional support for the list of paths:

```ts
fetchFile({ s3, bucket, path: ['fallback-path', 'most-probably-existing-file'] })
```

### CODE REVIEW

**Review by nl0 — CHANGES_REQUESTED** [review]
the implementation seems counter-intuitive for couple of reasons:
- only the first and the last items of the list are tried, other items possibly existing in the middle are ignored
- last item is tried first, which doesnt make sense to me

proposition: iterate thru the list from start and return the first positive result

File: catalog/app/containers/Bucket/requests.js [hunk 1]
```diff
@@ -218,8 +218,28 @@ export const bucketStats = async ({ req, s3, bucket, overviewUrl }) => {
...
+
+  return (
+    fileExists ||
+    (await ensureObjectIsPresentInCollection({ s3, bucket, key: R.init(key) }))
+  )
+}
+
 const fetchFileVersioned = async ({ s3, bucket, path, version }) => {
-  const versionExists = await ensureObjectIsPresent({
+  const isPresent = Array.isArray(path)
```
Comment (nl0) [inline catalog/app/containers/Bucket/requests.js:239]: we could just normalize path (convert it to a list) and use `ensureObjectIsPresentInCollection`

**Comment by fiskus** [comment]
> only the first and the last items of the list are tried, other items possibly existing in the middle are ignored

You probably confused `R.init` with `R.head`. `R.head` returns first element of array, but `R.init` returns _array_ without last element.

I double-checked: all items on the list are tried

**Comment by nl0** [comment]
> > only the first and the last items of the list are tried, other items possibly existing in the middle are ignored
> 
> You probably confused `R.init` with `R.head`. `R.head` returns first element of array, but `R.init` returns _array_ without last element.
> 
> I double-checked: all items on the list are tried

yeah i see now, but still i think it's more logical to use the natural order (try the first item of the array first)

**Comment by fiskus** [comment]
I turned the list of file paths over

File: catalog/app/containers/Bucket/requests.js [hunk 2]
```diff
@@ -218,11 +218,29 @@ export const bucketStats = async ({ req, s3, bucket, overviewUrl }) => {
...
 }
 
+// key is array of keys: string[]
+const ensureObjectIsPresentInCollection = async ({ s3, bucket, keys, version }) => {
+  if (!keys.length) return null
+
+  const fileExists = await ensureObjectIsPresent({
+    s3,
+    bucket,
+    key: R.head(keys),
```
Comment (nl0) [inline catalog/app/containers/Bucket/requests.js:228]: `keys[0]`? =)

**Review by nl0 — APPROVED** [review]

**Pull request merged**